core/lib/common/test.js

```diff
@@ -43,6 +43,25 @@ module.exports = class Test {
 
 		this.archiver = Archiver;
 		this.context = new Context(this.suite.context);
+
+		// Return a proxy of this object that will forward access to properties
+		// present in the context object to that object.
+		//
+		// This allows us to shorten, e.g.:
+		//
+		// this.context.get().worker.executeCommandInHostOS(...)
+		//
+		// To just:
+		//
+		// this.worker.executeCommandInHostOS(...)
+		return new Proxy(this, {
```

Comment:
Although great use of the Proxy method in JS (TIL for me) I feel we are skirting around the issue with this abstraction. 

Here are my thoughts, 
1. We reduce the complexity of "context" by using Tap's context (https://node-tap.org/docs/api/test-lifecycle-events/#tcontext) in our tests instead our own implementation (Happy to have a call about this)
2. My second one is getting resolved with core moving off to testbot. My issue was having operations like 

this.context.get().worker.executeCommandInHostOS(...) in tests and why not directly using something simple like `worker.executeCommandInHostOS` or using tap's context? Just dropping this in for discussion. I think this is a great change already, LGTM.

Reply:
I agree with the above completely.

Reply:
Can't do that because some things in the context object are populated mid test. 

Example in HUP test: The path of the downloaded image gets added to the context mid test. Well, we can download that image in beforeEach too tbh. So can be explored. I know some stuff definitely which gets configured in test (probably everything in suite.js comes under t.beforeEach()